core/src/raw/http_util/mod.rs

```diff
@@ -24,6 +24,7 @@
 
 mod client;
 pub use client::HttpClient;
+pub use client::CREDENTIAL_CLIENT;
```

Comment:
Please mark this as `pub(crate)`. I prefer not to expose it to our users, and it's highly likely that we'll remove it in the future.